enclave-runtime/src/lib.rs

```diff
@@ -749,13 +749,14 @@ where
 		};
 
 		// compose indirect block confirmation
+		// should be changed to ParentchainBlockProcessed, see worker issue #457
 		let xt_block = [TEEREX_MODULE, BLOCK_CONFIRMED];
-		let genesis_hash = validator.genesis_hash(validator.num_relays()).unwrap();
+		let mrenclave: ShardIdentifier = on_chain_ocall_api.get_mrenclave_of_self()?.m.into();
 		let block_hash = signed_block.block.header().hash();
 		let prev_state_hash = signed_block.block.header().parent_hash();
 		calls.push(OpaqueCall::from_tuple(&(
 			xt_block,
-			genesis_hash,
+			mrenclave, // pallet teerex only accepts shard == mrenclave for now
```

Comment:
`For now` is very ambiguous. I would like to see the following:

`mrenclave, // 'block_confirmed' does only accept shard == mrenclave. Change to 'genesis_hash' when #457 is tackled.`

Reply:
Is it? We want to have more than one shard available per enclave, otherwise we wouldn't need shards at all, so I don't think it's that ambigious. Also, will we need to re-add genesis hash? Since we will be using separate construct in #457 for parentchain block confirmations, we might as well change the delivered information.

Anyway, I'll adapt.

Reply:
Ok, agreed. We can change the latter part. I was mainly concerned about the for now part. A third party who reads that part has no chance to see what 'for now' means. You can change that to whatever you seem sensible, e.g. `will probably be adjusted when #457` is tackled.

Reply:
Changed to `
// 'block_confirmed' only accepts shard == mrenclave. Will be adjusted with #457`. Is that allright?

Reply:
Fine, thanks for coping with me. :)

Reply:
Ah -  that also gives off the wrong idea. Added :
`// 'block_confirmed' only accepts shard == mrenclave. Overall 'block_confirmed' construct will be adjusted with #457`